api/pkg/s3/objectdelete.go

```diff
@@ -28,7 +28,12 @@ func (s *APIService) ObjectDelete(request *restful.Request, response *restful.Re
 	url := request.Request.URL
 	bucketName := request.PathParameter("bucketName")
 	objectName := request.PathParameter("objectKey")
-	version := url.Query().Get("versionId")
+	versionId := request.Request.URL.Query().Get(common.REQUEST_QUERY_VERSION_ID)
+	if versionId != ""{
+		objectName = objectName + "_" + versionId
```

Comment:
Does this mean, for a specific object, if versioning is enabled, then it's key will be different from the one with versioning disabled?